actions/rules/action_field_set.go

```diff
@@ -0,0 +1,91 @@
+package rules
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/fabric8-services/fabric8-wit/application"
+	"github.com/fabric8-services/fabric8-wit/convert"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+)
+
+// ActionFieldSet takes a configuration JSON object that has field
```

Comment:
> ActionFieldSet takes a configuration JSON object

The struct itself does not contain anything like this.

Reply:
The JSON is still marshalled at this point.

Reply:
To Konrad's point, i think this is confusing, anyway we can improve this?

Reply:
@tinakurian there is a slight confusion here: the generic action infa defines that an action rule gets a configuration in the form of a `string`. What this string contains is completely up to the rule. It may be a JSON (like in this case), it may be a YAML or anything else. The reason why this is not interface{} is because the rule configuration is stored as a string in marshalled form and the idea is to not deal with the unmarshalling outside of the rule itself as only the rule knows what it is and doing the unmarshalling in an upper layer would possibly break the atomic concept of the rule, spreading parts of the rule "knowledge" in the form of a config or impl over the system. The rules need to be atomic as they may be provided by external sources later (for example the customer).